### DISCUSSION TITLE
[config] Add readinessProbe/livenessProbe checks to Kubernetes/Helm

### DISCUSSION THREAD

**Discussion by atombrella** [aquasecurity/trivy#8161] [Ideas]
### Description

I couldn't find this listed in https://www.stigviewer.com/stig/kubernetes/

Kubernetes' official documentation:
https://kubernetes.io/docs/concepts/configuration/liveness-readiness-startup-probes/

If you create a basic chart with `helm create sample` and then remove the `livenessProbe`, `readinessProbe` (and `startupProbe`), Trivy doesn't come with these suggestions.

It's related to reliability, and not security. The only Kubernetes linter that I know of which checks for this is 
[kube-score](https://github.com/zegl/kube-score/blob/master/README_PROBES.md)

```
Version: 0.58.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-10-05 12:15:24.045381837 +0000 UTC
  NextUpdate: 2024-10-06 12:15:24.045381707 +0000 UTC
  DownloadedAt: 2024-10-05 16:24:18.819563279 +0000 UTC
Check Bundle:
  Digest: sha256:ef2d9ad4fce0f933b20a662004d7e55bf200987c180e7f2cd531af631f408bb3
  DownloadedAt: 2024-12-23 11:47:10.385921794 +0000 UTC
```

If possible, you can leave some hints in this issue how to implement this. I guess `defsec` would be a good place to start.

### Target

Kubernetes

### Scanner

Misconfiguration

**Comment by nikpivkin** [comment]
Hi @atombrella !

This is not security related and may create noise for users. You can write a [custom check](https://trivy.dev/latest/docs/scanner/misconfiguration/custom/) that will check for the presence of `readinessProbe/livenessProbe`.

**Reply by atombrella** [reply]
OK. I'll try to write those 2/3 checks :) Maybe as a compromise, they could be added inside a contrib-folder, so they won't be part of the official bundle?

**Reply by nikpivkin** [reply]
what do you mean by 2/3 checks?

**Reply by itaysk** [reply]
Hi there, we are not going to add a contrib directory with non-security checks, since it might still create expectations that we maintain those checks. alternatively, feel free to create another repo for complementary trivy checks that you/the community can maintain. if you want to make it more discoverable, you can add it to the ecosystem section in the trivy documentation.

**Reply by atombrella** [reply]
OK. I got the checks working. I found this https://github.com/aquasecurity/trivy/discussions/7918 which is something that I wondered a bit when getting matches. Without `avd_id` Trivy just outputs this:

```
 (MEDIUM): Deployment is missing readinessProbe
══════════════════════════════════════════════════════════════════════════════════════════════════
Deployments must have livenessProbe, readinessProbe and startupProbe for reliability.
```

But with `avd_id` it looks nicer. https://trivy.dev/v0.58/tutorials/misconfiguration/custom-checks/ The documentation doesn't highlight this.

Please just close this issue/discussion. Seems like a nice idea to create a repository with some of these non-security, but reliability checks and link them. Probably I can find a few more, but the `readinessProbe`, `livenessProbe` and `startupProbe` are quite valuable suggestions.
